### PR TITLE
Support new event types

### LINKED ISSUE #215: Support new event types
The `type` column of the `event` table now has a defined type which only allows certain values:

- acknowledgement-cleared
- acknowledgement-set
- custom
- downtime-end
- downtime-removed
- downtime-start
- flapping-end
- flapping-start
- incident-age (previously internal)
- state

Previously only the following values where used:

- state
- internal (now incident-age)

We need to support the new values, remove all workarounds (`internal` support, message matching) and show proper phrases:

Type | Phrase
-------|-----------
acknowledgement-cleared | was unacknowledged
acknowledgement-set | has been acknowledged
custom | _no phrase needed_
downtime-end | left a downtime period
downtime-removed | prematurely left a downtime period
downtime-start | entered a downtime period
flapping-end | left a flapping period
flapping-start | entered a flapping period
incident-age | exceeded **a** time constraint
state | _depending on severity, unchanged_

### PR DESCRIPTION
**Pull request by sukhwinder33445**
resolves #215

### CODE REVIEW

File: library/Notifications/Common/Icons.php [hunk 1]
```diff
@@ -41,4 +41,32 @@ private function __construct()
...
+
+    public const FLAPPING = 'bolt';
+
+    public const INCIDENT_AGE = 'hourglass-end'; //TODO: icon? and for custom?
+
+    public const SEVERITY_OK = 'heart';
+
+    public const SEVERITY_CRIT = 'circle-exclamation';
+
+    public const SEVERITY_WARN = 'exclamation-triangle';
```
Comment (nilmerg) [inline library/Notifications/Common/Icons.php:59]: On fontawesome.com this is `triangle-exclamation`

File: library/Notifications/Common/Icons.php [hunk 2]
```diff
@@ -41,4 +41,32 @@ private function __construct()
...
+
+    public const SEVERITY_ERR = 'circle-xmark';
+
+    public const SEVERITY_DEBUG = 'bug-slash';
+
+    public const SEVERITY_INFO = 'info';
```
Comment (nilmerg) [inline library/Notifications/Common/Icons.php:65]: `circle-info`

File: library/Notifications/Common/Icons.php [hunk 3]
```diff
@@ -41,4 +41,32 @@ private function __construct()
...
+
+    public const ACKNOWLEDGED = 'check';
+
+    public const UNACKNOWLEDGED = 'xmark';
+
+    public const DOWNTIME = 'plug';
+
+    public const FLAPPING = 'bolt';
+
+    public const INCIDENT_AGE = 'hourglass-end'; //TODO: icon? and for custom?
```
Comment (nilmerg) [inline library/Notifications/Common/Icons.php:53]: for custom: `message`

File: library/Notifications/Common/Icons.php [hunk 4]
```diff
@@ -41,4 +41,32 @@ private function __construct()
...
+
+    public const SEVERITY_ERR = 'circle-xmark';
+
+    public const SEVERITY_DEBUG = 'bug-slash';
+
+    public const SEVERITY_INFO = 'info';
+
+    public const SEVERITY_ALERT = 'bell';
+
+    public const SEVERITY_EMERG = 'tower-broadcast';
```
Comment (nilmerg) [inline library/Notifications/Common/Icons.php:69]: I'd prefer `bullhorn`. It matches the general style of severity icons better. (Solid and bulky)

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Notifications/Common/Icons.php (hunk 1 above)
Comment (sukhwinder33445) [reply library/Notifications/Common/Icons.php:59]: They changed it on v6, it still works, but I will change it.

File: library/Notifications/Model/Event.php [hunk 5]
```diff
@@ -153,4 +155,112 @@ public static function mapSeverity(?string $severity): ?string
...
+                $icon = new Icon(Icons::DOWNTIME);
+                break;
+            case 'flapping-start':
+            case 'flapping-end':
+                $icon = new Icon(Icons::FLAPPING);
+                break;
+            case 'incident-age':
+                $icon = new Icon(Icons::INCIDENT_AGE);
+                break;
+            default:
```
Comment (nilmerg) [inline library/Notifications/Model/Event.php:260]: Please add a case here. I don't want this icon to be the default for any upcoming types.

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]

**Pull request merged**